src/App.js

```diff
@@ -1,73 +1,81 @@
+import ClassDialog from './components/ClassDialog.js'
+import HooksDialog from './components/HooksDialog.js'
+
+// These two constants function as enums in this example. The state variable `example_type` will
+// either be CLASS_TYPE or HOOKS_TYPE
+const CLASS_TYPE = 'class'
+const HOOKS_TYPE = 'hooks'
 
 // Every React class extends React.Component (this is like the basic React Object)
 class App extends Component {
   // Every Class has this thing called 'state' - it's what makes React so amazing.
   // Basically, React keeps track of whenever a state variable changes, and when it does
   // it re-renders anything necessary to show that change on the page. The alternative
-  // would be keeping track yourself
-  state = { open: false, name: '' }
+  // would be keeping track of everything that /could/ change all by yourself
+  state = { example_type: CLASS_TYPE }
 
   // This is the same as a private method in Java. You might see some people using a syntax like:
-  //  function handleClose(){...}
+  // function handleClick(){...}
   // This is a bit of personal preference, but typically fat-arrow functions are nicer to work
-  // with. This weird arrow function (shown below) binds the method //to this class, which is 
+  // with. This weird arrow function (shown below) binds the method to this class, which is
   // what most Java people are used to happening automatically.
-  hanldeClick = () => {
-    // To change state, you have to call the setState method
-    // It accepts an object as a parameter
-    this.setState({ open: !this.state.open })
-  }
-
-  // This method is called when the text field is changed. It passes an `event` which we can get the
-  // target value of
-  handleChange = event => {
-    this.setState({ name: event.target.value })
+  // Inside the method, we need to call this.setState() to change any variable we previously
+  // declared in state up above. this.setState is given to us because we extend Component
+  handleClick = event => {
+    this.setState({ example_type: event.target.value })
   }
 
-
-  // Because this class extends React.Component, it inherits several lifecylce methods.
-  // One that shows up in basically every class is `render` - this is where one typically
+  // Because this class extends React.Component, it inherits several lifecycle methods.
+  // One that shows up in basically every class is `render()` - this is where one typically
   // puts all of the JSX you want to see on the page
   render() {
+    // Instead of having to type `this.state.example_type`, we can use a shorter name in the
+    // render function. If you have lots of state variables, take a look at Object Destructuring
+    const type = this.state.example_type
+
+    // Depending on what the state variable `type` equals, our React will render different things
     return (
-      // There has to be one parent container - you can't render two top-level components
-      // In this case that countainer is a div
```

Comment:
I don't think that these comments should be removed

Reply:
Added back in :+1: